frontend/packages/console-shared/src/components/status/status.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import * as History from 'history';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import { Popover, PopoverPosition } from '@patternfly/react-core';
+import { Button } from 'patternfly-react';
+import { ErrorCircleOIcon, InProgressIcon, OkIcon, InfoAltIcon } from '@patternfly/react-icons';
+import {
+  global_danger_color_100 as dangerColor,
+  global_success_color_200 as successColor,
+} from '@patternfly/react-tokens';
+
+import { CamelCaseWrap } from '@console/internal/components/utils';
+import { DASH } from '../../constants';
+
+type StatusProps = {
+  icon?: React.ReactElement;
```

Comment:
Is the intention to only support patternfly react icons? Or any component? If it's only a patternfly react icon, we can change the prop types to accept `color` and `size` just like the pattern fly react icons, and instead of `ReactElement` accept a component type. Then you can render it without the need to `cloneElement`.